### ISSUE TITLE
Bug: Gas is Free and Unlimited

### ISSUE THREAD

**Issue by stubkan** [CleverRaven/Cataclysm-DDA#48422]
# Describe the bug

Vast amounts of gasoline and diesel are available for free at many gas stations dotted all over the map.  Cash cards etc, are not required to pay for them.

# Steps To Reproduce

1. Drive up to gas station.  
2. Help yourself to gas.
3. Help yourself to more.
4. Laugh at the idiots who left their empty-of-gas cars around the gas station, they were too stupid to get the free gas.

# Expected behavior

1. Gas was not going to be free
2. Perhaps the gas tanks would be empty
3. Or the tanks were draining
4. Gas would be scarce and precious - like in the Mad Max universe

I had been collecting cash cards in anticipation for having to pay for gas...  I found this was unnecessary, and there is no other important use for cash, so I no longer collect them.

Does not fit in with the apocalyptic setting, why didnt all the other people drain the gas stations in their panic? (notwithstanding that it was free)  Not very balancing, as once you get a car, you are set forever for gas with no effort required.

It does not take very much to drain a gas station of its stores irl, they are restocked weekly at a minimum to keep up with demand.  If demand flucuates slightly - such as a pandemic - often they can find them selves low on gas quite easily.  

Gas scarcity would also reflect a post-apocalyptic world better.  It would also contribute more to fun, in terms of scavenging and resource management especially in the early to mid game.

# Screenshots

# Versions and configuration

0.E to 0.F

# Additional context

**Comment by Night-Pryanik** [comment]
Could you please rewrite "steps to reproduce" to be more clear?

**Comment by PatrikLundell** [comment]
In 0.E2 stable I've seen two kinds of gas stations:
- Ones with a shop (where you pay inside, as Xenomorph-III mentioned), that may have both diesel and gasoline pumps. The diesel pumps I've seen have had at most 40 L in them, and thus I've never been able to fill my truck fully (until I found multiple barrels of diesel at a site). This fuel is "free" as the shops aren't manned anymore (and it's "free" in real life too, until the cops get you).
- Ones without a shop and payment is at the pump. I've never seen any of those providing diesel. They have a gasoline storage at the site, and I believe all the pumps get their gasoline from this.

**Comment by stubkan** [comment]
If the gas stations are surrounded by abandoned cars and it's 60 days after the apocalypse...  Any 'free' gas in the pumps would be long gone.  Besides, I've never seen a pump in my life that didnt require unlocking somehow - and I've travelled extensively in a few countries.  Even the old shit ones have physical locks and what have you not.

I'm on 0.E too, and every gas station I have found has free gas, I don't know if it's different now...

**Comment by PatrikLundell** [comment]
It used to be that you just drove up to the pump, filled the tank, and then went inside to pay. Later (I can't say when) they added activation from inside the shop so they manually "approve" of the filling, assuming cash is accepted at all nowadays. However, I've never been to New England, so I have no idea of where they are on that "upgrade" ladder.

**Comment by stubkan** [comment]
Yes, although, even the hi-tech gas stations have all 6 pumps full of free gas and diesel.  Why didnt all the panicking people in the 60 days of apocalypse find time to grab that free gas while they were drivin their cars down to empty tanks?

**Comment by Night-Pryanik** [comment]
Looks like interaction with gas pump was broken at some time as previously they all required a payment in advance.

**Comment by actual-nh** [comment]
How do the pumps have power to pump gas/diesel?

**Comment by Zireael07** [comment]
Option 1) suspension of disbelief for gameplay purposes (not having any way to acquire gas/diesel would suck)
Option 2) solar power

**Comment by actual-nh** [comment]
> Option 1) suspension of disbelief for gameplay purposes (not having any way to acquire gas/diesel would suck)

Open the caps on the underground tanks and siphon.

**Comment by PatrikLundell** [comment]
As far as I know siphoning from gas station tanks isn't supported, although it would make sense IRL.

It can also be noted that you can drive vehicles to a fairly decent extent by siphoning fuel from more or less wrecked vehicles (I've gotten a fair bit more diesel that way than from the few gas stations that had diesel pumps). In the longer term it's also possible to make bio diesel.

I gradually switched over to solar power (using battery swapping in addition to of reinforced solar panels on the roof), but as far as I understand changes have made that a difficult path since both solar panels and batteries get smashed up by propagated forces whenever you run over zombies (and repair uses up non renewable resources).

**Comment by actual-nh** [comment]
> As far as I know siphoning from gas station tanks isn't supported, although it would make sense IRL.

Supporting it would be good.

> In the longer term it's also possible to make bio diesel.

And re solar panels:
> repair uses up non renewable resources

Long-term, the player faction ending should at least partially be determined by how much you're depending on non-renewables. The IRREPLACEABLE_CONSUMABLE flag may help lay the foundation for that.

**Comment by RDru** [comment]
> 
> 
> Yes, although, even the hi-tech gas stations have all 6 pumps full of free gas and diesel. Why didnt all the panicking people in the 60 days of apocalypse find time to grab that free gas while they were drivin their cars down to empty tanks?

Seeing how many zombies are in cities I don't think anyone drove thought them and survived

**Comment by actual-nh** [comment]
> > Yes, although, even the hi-tech gas stations have all 6 pumps full of free gas and diesel. Why didnt all the panicking people in the 60 days of apocalypse find time to grab that free gas while they were drivin their cars down to empty tanks?
> 
> Seeing how many zombies are in cities I don't think anyone drove through them and survived

What about gas stations along roads in less-populated areas?

**Comment by LeahLuong** [comment]
@NetSysFire Night-Pryanik stated earlier in this thread:

> Looks like interaction with gas pump was broken at some time as previously they all required a payment in advance.

Shouldn't this issue retain the "Bug" tag?? Also, was it that tag that prevented stalebot from marking this issue?? If so, are there other tags w/ this effect??

**Comment by NetSysFire** [comment]
This looks like a regression and I honestly have no idea about stalebot, but there are other issues flagged with Exploit which stalebot is leaving alone. So it should just work™.

Unfortunately we do not have a tag for regressions and I am going through older issues seeing if they need any extra attention and Exploit seems way more appropriate than Bug. Feel free to correct me though.
